domain/src/test/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/AgencyTest.java

```diff
@@ -0,0 +1,151 @@
+
+        //noinspection ConstantConditions
+        underTest.agencyId(STRING_TEST_VALUE)
+                .agencyName(null)
+                .agencyUrl(STRING_TEST_VALUE)
+                .agencyTimezone(STRING_TEST_VALUE)
+                .agencyLang(STRING_TEST_VALUE)
+                .agencyPhone(STRING_TEST_VALUE)
+                .agencyFareUrl(STRING_TEST_VALUE)
+                .agencyEmail(STRING_TEST_VALUE);
+
+        final var data = underTest.build(mockNoticeCollection);
+
+        final ArgumentCaptor<MissingRequiredValueNotice> captor =
+                ArgumentCaptor.forClass(MissingRequiredValueNotice.class);
+
+        verify(mockNoticeCollection, times(1)).add(captor.capture());
+
+        final List<MissingRequiredValueNotice> noticeList = captor.getAllValues();
+
+        assertEquals("agency.txt", noticeList.get(0).getFilename());
+        assertEquals("agency_name", noticeList.get(0).getFieldName());
+        assertEquals(STRING_TEST_VALUE, noticeList.get(0).getEntityId());
+
+        assertTrue(data.getData() instanceof List);
+    }
+
+    // Field agencyUrl is annotated as `@NonNull` but test require this field to be null. Therefore annotation
+    // "@SuppressWarnings("ConstantConditions")" is used here to suppress lint.
+    @Test
+    public void createAgencyWithNullAgencyUrlShouldGenerateMissingRequiredValueNotice() {
+        @SuppressWarnings("unchecked") final List<Notice> mockNoticeCollection = mock(ArrayList.class);
+        final Agency.AgencyBuilder underTest = new Agency.AgencyBuilder();
+
+        //noinspection ConstantConditions
+        underTest.agencyId(STRING_TEST_VALUE)
+                .agencyName(STRING_TEST_VALUE)
+                .agencyUrl(null)
+                .agencyTimezone(STRING_TEST_VALUE)
+                .agencyLang(STRING_TEST_VALUE)
+                .agencyPhone(STRING_TEST_VALUE)
+                .agencyFareUrl(STRING_TEST_VALUE)
+                .agencyEmail(STRING_TEST_VALUE);
+
+        final var data = underTest.build(mockNoticeCollection);
+
+        final ArgumentCaptor<MissingRequiredValueNotice> captor =
+                ArgumentCaptor.forClass(MissingRequiredValueNotice.class);
+
+        verify(mockNoticeCollection, times(1)).add(captor.capture());
+
+        final List<MissingRequiredValueNotice> noticeList = captor.getAllValues();
+
+        assertEquals("agency.txt", noticeList.get(0).getFilename());
+        assertEquals("agency_url", noticeList.get(0).getFieldName());
+        assertEquals(STRING_TEST_VALUE, noticeList.get(0).getEntityId());
+
+        assertTrue(data.getData() instanceof List);
+    }
+
+    // Field agencyTimezone is annotated as `@NonNull` but test require this field to be null. Therefore annotation
+    // "@SuppressWarnings("ConstantConditions")" is used here to suppress lint.
+    @Test
+    public void createAgencyWithTimezoneAgencyUrlShouldGenerateMissingRequiredValueNotice() {
```

Comment:
Shall that be createAgencyWith**Null**TimezoneAgencyUrlShouldGenerateMissingRequiredValueNotice